types/msgservice/validate.go

```diff
@@ -0,0 +1,70 @@
+package msgservice
+
+import (
+	"fmt"
+
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoregistry"
+
+	msg "cosmossdk.io/api/cosmos/msg/v1"
+)
+
+// ValidateServiceAnnotations validates that all Msg services have the
+// `(cosmos.msg.v1.service) = true` proto annotation.
+func ValidateServiceAnnotations(serviceName string) error {
+	sd, err := protoregistry.GlobalFiles.FindDescriptorByName(protoreflect.FullName(serviceName))
```

Comment:
This needs the pulsar generated files to be registered. It means that app developer need to 1. generate their proto with pulsar 2. import them somewhere (e.g. with `_`). It seems like a high migration path.

What I propose:
- we don't throw an error here. If the descriptor is not found, we simply skip
- in this file, we import all SDK's descriptors as `_`, so at least the SDK's ones are validated.

This way, we'll only validate the imported descriptors. And app developers can do the transition to pulsar at their own pace.